base/intfuncs.jl

```diff
@@ -198,7 +198,7 @@ end
 function ndigits0z(n::Unsigned, b::Int)
     d = 0
     if b < 0
-        d = ndigits0znb(n, b)
+        d = ndigits0znb(signed(n), b)
```

Comment:
I think so, since after removing `signed` the unsigned wrapping triggered a failure. So I guess this means with `signed`, the wrapping is correct (since the test works). But you'll have to find somebody who knows this code to get an actual answer. :-)